lib/useOnyx.ts

```diff
@@ -57,8 +57,39 @@ type ResultMetadata = {
 
 type UseOnyxResult<TKey extends OnyxKey, TValue> = [CachedValue<TKey, TValue>, ResultMetadata];
 
-function getCachedValue<TKey extends OnyxKey, TValue>(key: TKey, selector?: Selector<TKey, unknown, unknown>): CachedValue<TKey, TValue> | undefined {
-    return (OnyxUtils.tryGetCachedValue(key, {selector}) ?? undefined) as CachedValue<TKey, TValue> | undefined;
+type UseOnyxSelector<TKey extends OnyxKey, SelectorValue> = (data?: OnyxValue<TKey>) => SelectorValue;
+
+function getCachedValue<TKey extends OnyxKey>(key: TKey): OnyxValue<TKey> | undefined {
+    if (!OnyxUtils.isCollectionKey(key)) {
+        return OnyxCache.get(key) as OnyxValue<TKey> | undefined;
+    }
+
+    const allCacheKeys = OnyxCache.getAllKeys();
+
+    // It is possible we haven't loaded all keys yet so we do not know if the
+    // collection actually exists.
+    if (allCacheKeys.size === 0) {
+        return;
+    }
+
+    const values: Record<string, unknown> = {};
+    allCacheKeys.forEach((cacheKey) => {
+        if (!cacheKey.startsWith(key)) {
+            return;
+        }
+
+        values[cacheKey] = OnyxCache.get(cacheKey);
+    });
+
+    return values as OnyxValue<TKey>;
```

Comment:
```suggestion
    return values as OnyxValue<TKey> | undefined;
```

Reply:
If I may use your help here - do you know in what circumstances Onyx collection could result to `undefined`? 

For this scope itself:

Value in this branch of the code (collection part) won't be set to `undefined` - it starts with empty object and is populated with collection items. 

`undefined` covers first branch for non-collection values where we could get undefined from cache.

Thanks for looking into it with such detail! <3